test/watch/test/file-watching.test.ts

```diff
@@ -63,3 +71,19 @@ test('editing config file reloads new changes', async () => {
     expect(vitest.getOutput()).toMatch('ok 2')
   })
 })
+
+describe('browser', () => {
+  test('editing source file triggers re-run', async () => {
```

Comment:
This test fails for some reason. I've been looking into it for a while but have no idea what's going on.

It works fine when run without `--browser` flags:

```sh
$ pnpm run test run --root fixtures

> @vitest/test-watch@ test /x/vitest/test/watch
> vitest "run" "--root" "fixtures"


 RUN  v0.29.7 /x/vitest/test/watch/fixtures

 ✓ math.test.ts (1)
   ✓ sum
 ✓ example.test.ts (1)
   ✓ getHello

 Test Files  2 passed (2)
```

But with `--browser` it does not respect `--root` when collecting tests. And somehow something ends up loading  test case from `floating-vue` dependency.

```sh
$ pnpm run test run --root fixtures --browser.enabled --browser.name=chrome

> @vitest/test-watch@ test /x/vitest/test/watch
> vitest "run" "--root" "fixtures" "--browser.enabled" "--browser.name=chrome"


 RUN  v0.29.7 /x/vitest/test/watch/fixtures
      Browser runner started at http://localhost:63315

1:58:28 PM [vite] Internal server error: Failed to resolve import "@peeky/test" from "../../node_modules/.pnpm/floating-vue@2.0.0-y.0_vue@3.2.45/node_modules/floating-vue/src/directives/v-tooltip.spec.ts?vitest=1679914708888". Does the file exist?

 ❯ ../node_modules/@vitest/browser/node_modules/@vitest/ui/node_modules/floating-vue/src/directives/v-tooltip.spec.ts (0)
 ✓ example.test.ts (1)
   ✓ getHello
 ❯ ../test/stdout.test.ts (0)
 ✓ math.test.ts (1)
   ✓ sum
 ❯ ../test/file-watching.test.ts (0)
 ❯ ../test/stdin.test.ts (0)

 Test Files  2 passed (2)
      Tests  2 passed (2)
   Start at  13:58:27
   Duration  2.19s (transform 0ms, setup 0ms, collect 111ms, tests 12ms)

 ELIFECYCLE  Test failed. See above for more details.
```

Reply:
Did you try making `root` an absolute file? `resolve(__dirname, 'fixtures')` for example

Reply:
It is probably bundled somehow and imported somewhere 🤔 Looking at `Test Files`, it says `2 passed (2)` - so only those were passed down to the browser, but others were imported from the outside somehow during the import of `math` and `example`

Reply:
The [adfaea2](https://github.com/vitest-dev/vitest/pull/3040/commits/adfaea20889d5575ea8f5c747f80a329f35addb7) fixes this.